src/transformers/models/whisper/tokenization_whisper.py

```diff
@@ -0,0 +1,624 @@
+
+        if language is not None:
+            language = language.lower()
+            if language not in LANGUAGES:
+                if language in TO_LANGUAGE_CODE:
+                    language = TO_LANGUAGE_CODE[language]
+                else:
+                    raise ValueError(f"Unsupported language: {language}")
+
+        if multilingual:
+            task = task or "transcribe"
+            language = language or "en"
+        else:
+            task = None
+            language = None
+
+        self.language = language
+
+        translate = self.all_special_ids[-6]
+        transcribe = self.all_special_ids[-5]
+        sot_sequence = [self.all_special_ids[1]]
+
+        if language is not None:
+            additional_tokens = dict(
+                zip(
+                    self.additional_special_tokens,
+                    self.additional_special_tokens_ids,
+                )
+            )
+            self.language_token = additional_tokens[f"<|{self.language}|>"]
+            langs = tuple(LANGUAGES.keys())
+            sot_sequence.append(self.all_special_ids[1] + 1 + langs.index(language))
+
+        if task is not None:
+            sot_sequence.append(transcribe if task == "transcribe" else translate)
+        self.sot_sequence = sot_sequence
+
+    @property
+    @lru_cache()
+    def non_speech_tokens(self) -> Tuple[int]:
+        """
+        Returns the list of tokens to suppress in order to avoid any speaker tags or non-speech annotations, to prevent
+        sampling texts that are not actually spoken in the audio, e.g.
+
+        - ♪♪♪
+        - ( SPEAKING FOREIGN LANGUAGE )
+        - [DAVID] Hey there,
+
+        keeping basic punctuations like commas, periods, question marks, exclamation points, etc.
+        """
+
+        result = set()
+        symbols = list("'\"#()*+-/:;<=>@[\\]^_`{|}~「」『』")
+        symbols += "<< >> <<< >>> -- --- -( -[ (' (\" (( )) ((( ))) [[ ]] {{ }} ♪♪ ♪♪♪".split()
+
+        # symbols that may be a single token or multiple tokens depending on the tokenizer.
+        # In case they're multiple tokens, suppress the first token, which is safe because:
+        # These are between U+2640 and U+267F miscellaneous symbols that are okay to suppress
+        # in generations, and in the 3-byte UTF-8 representation they share the first two bytes.
+        miscellaneous = set("♩♪♫♬♭♮♯")
+        assert all(0x2640 <= ord(c) <= 0x267F for c in miscellaneous)
+
+        for symbol in symbols + list(miscellaneous):
+            for tokens in [self.encode(symbol), self.encode(" " + symbol)]:
+                if len(tokens) == 1 or symbol in miscellaneous:
+                    result.add(tokens[0])
+
+        return tuple(sorted(result))
+
+    def _get_single_token_id(self, text) -> int:
+        tokens = self.encode(text)
+        return tokens[0]
+
+    @property
+    @lru_cache()
+    def eot(self) -> int:
+        return self.tokenizer.eos_token_id
```

Comment:
there is no `self.tokenizer` I think - think we can clean up most of this properties here (remove them)